### PR TITLE
fix(linter): detect typescript eslint alias rules

### LINKED ISSUE #7233: linter: no matching rules for some typescript rules
Hi team, I ran oxlint and I caught some issues:
```
! The following rules do not match the currently supported rules:
  | typescript/no-dupe-class-members
  | typescript/no-useless-constructor
  | typescript/no-loss-of-precision
  | typescript/no-array-constructor
  | typescript/no-redeclare
  | typescript/no-unused-vars
```
But, you can see all these rules have been implemented. I am using  `"oxlint": "^0.11.0",`

_Originally posted by @markEHVN in https://github.com/oxc-project/oxc/issues/2180#issuecomment-2467175868_

### PR DESCRIPTION
**Pull request by Sysix**
closes #7233

### CODE REVIEW

**Comment by Sysix** [comment]
I do not have a clue why the NAPI test are failing. I hope this is just a CI problem,  
locally `pnpm run build && pnpm run test` works fine

File: crates/oxc_linter/src/config/rules.rs [hunk 1]
```diff
@@ -155,6 +155,10 @@ fn transform_rule_and_plugin_name<'a>(
         return (rule_name, "jest");
     }
 
+    if plugin_name == "typescript" && is_eslint_rule_adapted_to_typescript(rule_name) {
+        return (rule_name, "eslint");
+    }
```
Comment (camc314) [inline crates/oxc_linter/src/config/rules.rs:160]: question, so https://github.com/oxc-project/oxc/blob/6dd71c6d050630fe3ee29728f906884ab627987a/crates/oxc_linter/src/rules/typescript/no_unused_expressions.rs#L1

no-unused-expressions exists in `typescript` should it be moved to eslint?

Comment (Sysix) [reply crates/oxc_linter/src/config/rules.rs:160]: good catch. feel free to create an PR.

This should also fix the docs in #2180

Comment (camc314) [reply crates/oxc_linter/src/config/rules.rs:160]: thanks

https://github.com/oxc-project/oxc/pull/7624

**Review by camc314 — APPROVED** [review]

**Pull request merged**